examples/cluster/plot_hdbscan.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+"""
+====================================
+Demo of HDBSCAN clustering algorithm
```

Comment:
It would be good to visualize the properties specific to HDBSCAN. For example, the `probabilities_` attribute

Or the influence of `min_cluster_size` or `min_samples` on the clustering.

Reply:
Updated the plots to demonstrate effects of `min_cluster_size` and `min_samples` on the clustering and modified markers to more clearly demonstrate whether the point is core vs noise, and sized the markers proportional to their corresponding probabilities.